src/test/java/cloud/fogbow/accs/core/datastore/services/RecordServiceTest.java

```diff
@@ -0,0 +1,156 @@
+package cloud.fogbow.accs.core.datastore.services;
+
+import cloud.fogbow.accs.constants.Messages;
+import cloud.fogbow.accs.core.datastore.orderstorage.RecordRepository;
+import cloud.fogbow.accs.core.exceptions.InvalidIntervalException;
+import cloud.fogbow.accs.core.models.AccountingUser;
+import cloud.fogbow.accs.core.models.Record;
+import cloud.fogbow.accs.core.models.UserIdentity;
+import cloud.fogbow.accs.core.models.orders.OrderState;
+import cloud.fogbow.common.models.SystemUser;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.powermock.modules.junit4.PowerMockRunnerDelegate;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.sql.Timestamp;
+import java.text.ParseException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@PowerMockIgnore({"javax.management.*"})
+@PrepareForTest({RecordRepository.class, RecordService.class})
+@RunWith(PowerMockRunner.class)
+@PowerMockRunnerDelegate(SpringRunner.class)
+@SpringBootTest
+public class RecordServiceTest extends ServicesBaseUnitTest{
+
+    private final String ANY_VALUE = "any";
+    private final String TEST_DATE = "2000-01-01";
+    private final String DEFAULT_RESOURCE_TYPE = "compute";
+
+    //test case: When the start time is greater than the end time the
+    //message checked below must be part of the threw exception.
+    @Test
+    public void testCheckIntervalWhenStartTimeIsGreaterThanAllowed() {
+        //setup
+        Timestamp endTimeStamp = new Timestamp(System.currentTimeMillis());
+        Timestamp startTimeStamp = new Timestamp(System.currentTimeMillis() + 1);
+
+        try {
+            //exercise
+            recordService.checkInterval(startTimeStamp, endTimeStamp);
```

Comment:
Use 'Assert.fail()' after line 48 to evidence that the test must fail on this 'try/catch' line.